c7n/resources/ec2.py

```diff
@@ -2199,3 +2202,46 @@ class resource_type(query.TypeInfo):
         date = 'AllocationTime'
         cfn_type = config_type = 'AWS::EC2::Host'
         permissions_enum = ('ec2:DescribeHosts',)
+
+
+class InstanceImageMixin(InstanceImageBase):
+    """
+    CELFilter Mixin class to provide InstanceImageBase to CEL
+    """
+
+    def get_instance_image(self, resource):
+        """
+        get_instance_image retrieves the image id from the provided resource
+        :param resource:
+        :return image:
+        """
+        image = super().get_instance_image(resource)
+        return image
+
+
+@filters.register('cel')
+class CELFilter(BaseCELFilter, InstanceImageMixin):
+    """
+    EC2-specific implementation of CELFilter
+    """
+
+    def get_permissions(self):
+        return ('ec2:DescribeInstanceAttribute',)
+
+    def process(self, resources, event=None, filter=Filter):
+        super().prefetch_instance_images(resources)
+
+        filtered_resources = []
+        for resource in resources:
+            cel_prgm = self.cel_env.program(self.cel_ast, functions=celpy.c7nlib.FUNCTIONS)
+            cel_activation = {
+                "Resource": celpy.json_to_cel(resource),
```

Comment:
is there a reason where upper casing? lower casing seems more keyboard friendly ;-)